youtube-client-app/src/app/youtube/services/youtube-api.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import {
+  forkJoin, map, Observable, Subject,
+} from 'rxjs';
+import { ISearchItem, ISearchResultItem } from '../../shared/models/search-items.models';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class YoutubeApiService {
+  private pathForSearch: string = 'search?';
+
+  private pathForVideos: string = 'videos?';
+
+  public searchItems$ = new Subject<ISearchItem[]>();
+
+  public isLoading$ = new Subject<boolean>();
+
+  constructor(private http: HttpClient) { }
+
+  private getSearchListByQuery(query: string): Observable<ISearchResultItem[]> {
+    return this.http.get(this.pathForSearch, {
+      params: {
+        type: 'video',
+        part: 'snippet',
+        maxResults: '20',
+        q: query,
+      },
+    })
+      .pipe(map((responce: any) => responce.items));
```

Comment:
you can use https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker or similar extension to avoid typos. 

You can use generic in http.get method to specify the return type
`this.http.get<Item[]>`

Reply:
also it is usually never a good idea to use `any` type